pkg/build/api/types.go

```diff
@@ -76,19 +87,71 @@ type BuildSpec struct {
 	// Output describes the Docker image the Strategy should produce.
 	Output BuildOutput
 
-	// Compute resource requirements to execute the build
+	// Resources computes resource requirements to execute the build.
 	Resources kapi.ResourceRequirements
 
 	// PostCommit is a build hook executed after the build output image is
 	// committed, before it is pushed to a registry.
 	PostCommit BuildPostCommitSpec
 
-	// Optional duration in seconds, counted from the time when a build pod gets
-	// scheduled in the system, that the build may be active on a node before the
-	// system actively tries to terminate the build; value must be positive integer
+	// CompletionDeadlineSeconds is an optional duration in seconds, counted from
+	// the time when a build pod gets scheduled in the system, that the build may
+	// be active on a node before the system actively tries to terminate the
+	// build; value must be positive integer.
 	CompletionDeadlineSeconds *int64
 }
 
+// BuildTriggerCause holds information about the trigger of a build.
+type BuildTriggerCause struct {
+	// Reason is used to store a human readable reason for why the build was
+	// triggered. E.g.: "Build manually triggered by user", "Build triggered by
+	// ConfigChange update", "Build triggered by ImageChange for image:
+	// FOO:BAR", etc.
+	Reason string
```

Comment:
i guess by your definition it's a message, but the intent was "this is the reason a build was run".
saying "this is the message the build was run" doesn't make as much sense to me.

if we can't use reason because it's reserved for very specific usage, i guess we can call it a message, but it seems to hamstring us from using the right name for the field to represent what it is.